File: sdk/ruby/nextpay_client.rb
require 'net/http'
require 'json'

class PayForgeClient
  def initialize(api_key, base_url = 'https://api.payforge.local/api/v1')
    @api_key = api_key
    @base_url = base_url.sub(/\/$/, '')
  end

  def create_payment(amount:, currency:, metadata: {})
    request('POST', '/payments', { amount: amount, currency: currency, metadata: metadata })
  end

  private

  def request(method, path, payload = nil)
    uri = URI("#{@base_url}#{path}")
    http = Net::HTTP.new(uri.host, uri.port)
    http.use_ssl = uri.scheme == 'https'
    req = Net::HTTP::Post.new(uri)
    req['content-type'] = 'application/json'
    req['x-api-key'] = @api_key
    req.body = payload.to_json if payload

    res = http.request(req)
    raise "payforge_request_failed" unless res.code.to_i < 400

    JSON.parse(res.body)
  end
end

File: sdk/python/nextpay_client.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Any, Dict, Optional
import requests


@dataclass
class PayForgeClient:
    api_key: str
    base_url: str = "https://api.payforge.local/api/v1"

    def _request(self, method: str, path: str, payload: Optional[Dict[str, Any]] = None) -> Any:
        headers = {
            "content-type": "application/json",
            "x-api-key": self.api_key,
        }
        response = requests.request(method, f"{self.base_url.rstrip('/')}{path}", json=payload, headers=headers, timeout=30)
        data = response.json() if response.content else {}
        if not response.ok:
            raise RuntimeError(data.get("error", "payforge_request_failed"))
        return data

    def create_payment(self, amount: int, currency: str, customer_id: Optional[str] = None, metadata: Optional[Dict[str, str]] = None) -> Any:
        body = {
            "amount": amount,
            "currency": currency,
            "customerId": customer_id,
            "metadata": metadata or {},
        }
        return self._request("POST", "/payments", body)

    def get_payment(self, payment_id: str) -> Any:
        return self._request("GET", f"/payments/{payment_id}")

    def refund_payment(self, payment_id: str, amount: int, reason: Optional[str] = None) -> Any:
        return self._request("POST", f"/payments/{payment_id}/refund", {"amount": amount, "reason": reason})
